### PR TITLE
Removed magic numbers

### PR DESCRIPTION
**Pull request by peiva-git**
While removing some magic numbers, I noticed that the allowed `Position` coordinates should also have some limitations due to how the `Position` class is implemented. I added some tests to reflect the desired behavior and corrected the constructor code accordingly.

### CODE REVIEW

File: core/src/it/units/sdm/project/board/Position.java [hunk 1]
```diff
@@ -10,21 +10,32 @@
  * a row index and a column index. Both of them must be a positive integer number.
  */
 public class Position implements Comparable<Position> {
+
+    /**
+     * Maximum allowed {@link Position} column index for this implementation.
+     * Since {@link Position}s are printed using chess-board-like coordinates, i.e. A5, G7 and so on,
+     * this limit is set to the size of the English alphabet - 1 to prevent incorrect string conversion
+     */
+    static final int MAXIMUM_COLUMN_INDEX = 25;
```
Comment (manuelkoso) [inline core/src/it/units/sdm/project/board/Position.java:19]: Missing visibility keyword

File: core/src/it/units/sdm/project/game/gui/screens/GameScreen.java [hunk 2]
```diff
@@ -15,12 +15,27 @@
...
 /**
  * Main {@link FreedomGame} screen. Should be displayed after the players and the board are set.
  */
 public class GameScreen implements Screen {
+    /**
+     * This needs to be set at least to {@code TILE_SIZE * NUMBER_OF_COLUMNS + LOG_AREA_MINIMUM_WIDTH}
+     * to prevent parts of the GUI from being cut out if the viewport.
+     * This constant defines the width of the viewport in game-world coordinates
+     */
+    static final int GAME_SCREEN_WORLD_WIDTH = TILE_SIZE * NUMBER_OF_COLUMNS + 300;
```
Comment (manuelkoso) [inline core/src/it/units/sdm/project/game/gui/screens/GameScreen.java:31]: Missing visibility keyword

File: core/src/it/units/sdm/project/game/gui/screens/GameScreen.java [hunk 3]
```diff
@@ -15,12 +15,27 @@
...
+    /**
+     * This needs to be set at least to {@code TILE_SIZE * NUMBER_OF_ROWS}
+     * to prevent parts of the GUI from being cut out of the viewport
+     * This constant defines the height of the viewport in game-world coordinates
+     */
+    static final int GAME_SCREEN_WORLD_HEIGHT = TILE_SIZE * NUMBER_OF_ROWS + 50;
```
Comment (manuelkoso) [inline core/src/it/units/sdm/project/game/gui/screens/GameScreen.java:37]: Missing visibility keyword

**Review by manuelkoso — CHANGES_REQUESTED** [review]
There are missing some visibility keywords.

**Comment by manuelkoso** [comment]
Add `protected` instead

**Comment by peiva-git** [comment]
With`protected` I don't think they're visible ro other classes in the same package

**Review by manuelkoso — APPROVED** [review]

**Pull request merged**